servant-mock/src/Servant/Mock.hs

```diff
@@ -139,6 +143,22 @@ instance (KnownSymbol s, HasMock rest) => HasMock (QueryFlag s :> rest) where
 instance (KnownSymbol h, FromHttpApiData a, HasMock rest) => HasMock (Header h a :> rest) where
   mock _ = \_ -> mock (Proxy :: Proxy rest)
 
+instance (HasMock rest, AuthData authdata, Arbitrary usr)
```

Comment:
I'm actually not all that happy with the current mock behaviour.   I'm gonna have a new look at it. But i'll document it in the process =)